src/main/java/org/hawkular/client/android/activity/MetricDataActivity.java

```diff
@@ -0,0 +1,167 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.activity;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+
+import org.hawkular.client.android.R;
+import org.hawkular.client.android.backend.BackendClient;
+import org.hawkular.client.android.backend.model.Metric;
+import org.hawkular.client.android.backend.model.MetricData;
+import org.hawkular.client.android.backend.model.Tenant;
+import org.hawkular.client.android.util.Intents;
+import org.hawkular.client.android.util.ViewDirector;
+import org.jboss.aerogear.android.pipe.callback.AbstractActivityCallback;
+
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+import lecho.lib.hellocharts.model.Axis;
+import lecho.lib.hellocharts.model.AxisValue;
+import lecho.lib.hellocharts.model.Line;
+import lecho.lib.hellocharts.model.LineChartData;
+import lecho.lib.hellocharts.model.PointValue;
+import lecho.lib.hellocharts.model.Viewport;
+import lecho.lib.hellocharts.view.LineChartView;
+import timber.log.Timber;
+
+public class MetricDataActivity extends AppCompatActivity {
+    @InjectView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @InjectView(R.id.chart)
+    LineChartView chart;
+
+    @Override
+    protected void onCreate(Bundle state) {
+        super.onCreate(state);
+        setContentView(R.layout.activity_chart);
+
+        setUpBindings();
+
+        setUpToolbar();
+
+        setUpMetricData();
+    }
+
+    private void setUpBindings() {
+        ButterKnife.inject(this);
+    }
+
+    private void setUpToolbar() {
+        setSupportActionBar(toolbar);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    private void setUpMetricData() {
+        showProgress();
+
+        BackendClient.getInstance().getMetricData(getTenant(), getMetric(), this, new MetricDataCallback());
+    }
+
+    private void showProgress() {
+        ViewDirector.of(this).using(R.id.animator).show(R.id.progress);
+    }
+
+    private Tenant getTenant() {
+        return getIntent().getParcelableExtra(Intents.Extras.TENANT);
+    }
+
+    private Metric getMetric() {
+        return getIntent().getParcelableExtra(Intents.Extras.METRIC);
+    }
+
+    private void setUpMetricData(List<MetricData> metricDataList) {
+        List<PointValue> chartPoints = new ArrayList<>();
+
+        List<AxisValue> chartAxisPoints = new ArrayList<>();
+
+        for (int metricDataPosition = 0; metricDataPosition < metricDataList.size(); metricDataPosition++) {
```

Comment:
In the screen shot, the order (by time) seems to be reversed (02:44 at the left, 02:35 on the right).
You should perhaps sort the data here, as you should probably not be sure that the server always returns data sorted by timestamp.